ddtrace/internal/writer/writer.py

```diff
@@ -496,13 +479,14 @@ def __init__(
 
         buffer_size = buffer_size or config._trace_writer_buffer_size
         max_payload_size = max_payload_size or config._trace_writer_payload_size
-        try:
-            client = WRITER_CLIENTS[self._api_version](buffer_size, max_payload_size)
-        except KeyError:
-            raise ValueError(
-                "Unsupported api version: '%s'. The supported versions are: %r"
-                % (self._api_version, ", ".join(sorted(WRITER_CLIENTS.keys())))
+        if self._api_version not in WRITER_CLIENTS:
+            log.warning(
+                "Unsupported api version: '%s'. The supported versions are: %r",
+                self._api_version,
+                ", ".join(sorted(WRITER_CLIENTS.keys())),
             )
+            self._api_version = sorted(WRITER_CLIENTS.keys())[-1]
+        client = WRITER_CLIENTS[self._api_version](buffer_size, max_payload_size)
```

Comment:
Should we still wrap this with a `try/except`?

Reply:
I don't think it's necessary because we are certain that `_api_version` is a valid key in `WRITER_CLIENTS` at this point.